ctms/acoustic_service.py

```diff
@@ -259,8 +259,10 @@ def _main_table_converter(self, contact, main_fields):
                             )
                         # TODO: These are boolean values on our models, but in Acoustic
                         # they are configured as text values where we record `True`
-                        # and `False` as `1` and `0` respectively. We should
-                        # configure Acoustic to use a boolean column for this data.
+                        # and `False` as `1` and `0` respectively.
+                        # Note that `amo_*` and `fxa_*` fields are omitted when contact does not
+                        # have these relations.
```

Comment:
Should we mention explicitly that those values will be `null`? For example, we _could_ have written this code so that if the contact didn't have a relation, we'd record the acoustic value as `0`.

I think that's technically new behavior too for the main table columns. As we saw when looking in Acoustic yesterday, `double_opt_in` and `has_opted_out_of_email` were `null` in some places.

Reply:
They will be `null` because no default is configured on the field in Acoustic. But someone could change the default to `no` and the code comment becomes obsolete. 
<img width="431" alt="Screenshot 2023-08-31 at 15 05 18" src="https://github.com/mozilla-it/ctms-api/assets/546692/8d12f7a9-e753-4be7-bde7-67c00e90e465">

Shall we mention that it relies on the Acoustic field default value? Is it worth it?

The subfields `amo` and `fxa` are skipped above in the iteration https://github.com/mozilla-it/ctms-api/blob/650d68e9b0fcbf2e83c8abcf41caa2bdf52e8568/ctms/acoustic_service.py#L237 
Without more changes, we can't set `0` there.